test/unit/iso_generator_test.rb

```diff
@@ -0,0 +1,20 @@
+require 'test_plugin_helper'
+
+class ForemanBootdisk::IsoGeneratorTest < ActiveSupport::TestCase
+  include ForemanBootdiskTestHelper
+  setup :setup_bootdisk
+  setup :setup_org_loc
+  setup :setup_subnet
+  setup :setup_host
+
+  setup do
+    @host.build = true
+  end
+
+  test "full host image generation generates via PXELinux type" do
+    @host.expects(:generate_pxe_template).with(:PXELinux)
```

Comment:
Well my goal was to test only PXELinux option in `generate_pxe_template`. Anyway, if I understand you correctly, you want this test to be complete so changed it accordingly.

Also added two more tests on the filename part, which looks quite fragile.